### PR TITLE
 Add "PRO" button to Elevation: map center widget

### PR DESCRIPTION
**Pull request by nnngrach**
[issue](https://github.com/osmandapp/OsmAnd-Issues/issues/2658)

Screenshot 2024-10-14 at 23 27 47 Screenshot 2024-10-14 at 23 14 33

https://github.com/user-attachments/assets/f7387609-4bcb-4597-9d31-e3d96d6dc769

### CODE REVIEW

File: Sources/Controllers/Map/Widgets/OAMapWidgetRegistry.mm [hunk 1]
```diff
@@ -351,9 +351,8 @@ - (OAMapWidgetInfo *)getWidgetInfoForType:(OAWidgetType *)widgetType
             BOOL passAvailable = !availableMode || [OAWidgetsAvailabilityHelper isWidgetAvailableWithWidgetId:widget.key appMode:appMode];
             BOOL defaultAvailable = !defaultMode || !widget.isCustomWidget;
             BOOL passMatchedPanels = !matchingPanelsMode || [panels containsObject:widget.widgetPanel];
-            BOOL passTypeAllowed = [widget getWidgetType] == nil || [[widget getWidgetType] isAllowed];
```
Comment (tigrim) [inline Sources/Controllers/Map/Widgets/OAMapWidgetRegistry.mm:354]: Is this feature already implemented on Android? We are synchronizing the flow of widgets with Android. I see they have such code as well

**Comment by nnngrach** [comment]
video after code updating. still works

https://github.com/user-attachments/assets/3473bc74-ea78-434f-9b20-3eef01c1a093

File: Sources/Controllers/Map/Widgets/WidgetType.swift [hunk 2]
```diff
@@ -110,7 +104,7 @@ class WidgetType: NSObject {
     }
 
     func isPurchased() -> Bool {
-        if (WidgetType.getProWidgets().contains(where: { $0 == self } )) {
+        if (WidgetType.getProWidgets().contains(where: { $0.id == self.id } )) {
```
Comment (tigrim) [inline Sources/Controllers/Map/Widgets/WidgetType.swift:107]: remove brackets

File: Sources/Controllers/Map/Widgets/WidgetType.swift [hunk 3]
```diff
@@ -168,7 +162,7 @@ class WidgetType: NSObject {
     }
 
     static func getProWidgets() -> [WidgetType] {
-        return [.elevationProfile];
+        return [.elevationProfile, .altitudeMapCenter];
```
Comment (tigrim) [inline Sources/Controllers/Map/Widgets/WidgetType.swift:165]: remove return

File: Sources/Controllers/Settings/WidgetSettings/WidgetGroupItemsViewController.swift [hunk 4]
```diff
@@ -45,21 +46,35 @@ class WidgetGroupItemsViewController: OABaseNavbarViewController {
             if let cell {
                 cell.titleLabel.text = item.title
                 cell.leftIconView.image = UIImage(named: item.iconName ?? "")
+                
+                cell.accessoryView = nil
+                if let widgetType = item.obj(forKey: "widget_type") as? WidgetType, !widgetType.isPurchased() {
+                    cell.accessoryView = UIImageView(image: UIImage.icPaymentLabelPro)
```
Comment (tigrim) [inline Sources/Controllers/Settings/WidgetSettings/WidgetGroupItemsViewController.swift:52]: UIImageView(image: .icPaymentLabelPro)

File: Sources/Controllers/Settings/WidgetSettings/WidgetGroupListViewController.swift [hunk 5]
```diff
@@ -305,6 +305,13 @@ extension WidgetGroupListViewController {
                 cell.leftIconView.image = UIImage(named: item.iconName ?? "")
                 cell.titleLabel.text = item.title
                 cell.valueLabel.text = item.descr
+                
+                cell.accessoryView = nil
+                let widgetGroup = item.obj(forKey: "widget_group") as? WidgetGroup
+                if let widgetType = item.obj(forKey: "widget_type") as? WidgetType, !widgetType.isPurchased(), widgetGroup == nil {
+                    cell.accessoryView = UIImageView(image: UIImage.icPaymentLabelPro)
```
Comment (tigrim) [inline Sources/Controllers/Settings/WidgetSettings/WidgetGroupListViewController.swift:312]: up

**Review by tigrim — APPROVED** [review]

**Pull request merged**